File: ccif/src/java/other/org/himalay/ccif/APDUHeader.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class APDUHeader extends BinStruct implements PublicBinMsg {




    // members variables
    // messageType
    public int messageType ;

    public APDUHeader () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize messageType

    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read messageType
        {
            messageType= BinPrimitive.readUI24(istream);
            retVal += 3;
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;



        // write messageType
        {
            ostream.writeByte((messageType  & 0x00FF0000)>>16);
            ostream.writeShort((messageType & 0x0000FFFF));
            retVal+=3;
        }
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("APDUHeader\n");
        dc.increaseIndent();
        int retVal= 0;
        // write messageType
        dc.indent();
        dc.getPs().println("messageType="+messageType+"(0x"+ Integer.toHexString(messageType)+")") ;
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for messageType
    public int getMessageType()
    {
        return messageType ;
    }


    // Setter for messageType
    public void setMessageType(int val)
    {
        this.messageType= val;
    }


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/APDUHeader.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:11 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "APDUHeader.h"

#ifdef org_himalay_ccif_APDUHeader__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
APDUHeader::APDUHeader() {
    references= 0;


    // messageType
    messageType= (ui24)0;
}
// Destructor
APDUHeader::~APDUHeader() {
    // messageType

}

int APDUHeader::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // messageType
    {
        int i1= (0x000000FF & ((int)istream.readUI8())) << 16;
        int i2= (0x0000FFFF & ((int)istream.readUI16()));
        messageType= i1 | i2;
        retVal += 3;
    }
    return retVal;
}

int APDUHeader::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read messageType
    {
        int i1= (0x000000FF & ((int)istream.readUI8())) << 16;
        int i2= (0x0000FFFF & ((int)istream.readUI16()));
        messageType= i1 | i2;
        retVal += 3;
    }

    return retVal;
}

int APDUHeader::write(DataOutputStream& ostream) { // throws IOException


    int retVal= 0;
    // write messageType
    {
        ostream.writeUI8((messageType  & 0x00FF0000)>>16);
        ostream.writeUI16((messageType & 0x0000FFFF));
        retVal+=3;
    }

    return retVal;
}

int APDUHeader::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("APDUHeader\n");
    dc.increaseIndent();
    int retVal= 0;
    // write messageType
    dc.indent();
    dc.getPs().print("messageType=");
    dc.getPs().println((long)messageType);
    dc.decreaseIndent();
    return retVal;
}


// Getter for messageType
ui24 APDUHeader::getMessageType()
{
    return this->messageType;
}

// Setter for messageType
void APDUHeader::setMessageType(ui24 val)
{
    this->messageType= val;
}

int APDUHeader::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_APDUHeader__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::APDUHeader* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::APDUHeader* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code